addons/account/models/partner.py

```diff
@@ -545,6 +545,9 @@ def _inverse_use_partner_credit_limit(self):
             if not partner.use_partner_credit_limit:
                 partner.credit_limit = self.env['ir.property']._get('credit_limit', 'res.partner')
 
+    # 'show_credit_limit' does not actually depend on 'name',
+    # but the @api.depends is there to make sure the compute is triggered on partner creation.
+    @api.depends('name')
```

Comment:
what about doing something more conventional like

```python
      @api.model
      def default_get(self, default_fields):
          values = super().default_get(default_fields)
          values['show_credit_limit'] = self.env.company.account_use_credit_limit
          return values
```
or simply defining the default in the field definition:
```python 
default=lambda self: self.env.company.account_use_credit_limit
```